common/SmokeTests/SmokeTest/SmokeTest.csproj

```diff
@@ -38,7 +38,7 @@
     <PackageReference Include="System.Net.NameResolution" Version="4.3.0" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp3.1'">
+  <ItemGroup Condition="$([MSBuild]::VersionLessThan('$(TargetFrameworkVersion)', '3.1'))">
```

Comment:
Wouldn't `net461` come out as a higher version than 3.1?

Reply:
You're right. After testing around with the following, `$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp3.1'))` seems better:

```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003" DefaultTargets="Test">
    <PropertyGroup>
        <TargetFramework Condition="'$(TargetFramework)' == ''">net7.0</TargetFramework>
        <TargetFrameworkIdentifier>$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)'))</TargetFrameworkIdentifier>
        <TargetFrameworkVersion>$([MSBuild]::GetTargetFrameworkVersion('$(TargetFramework)'))</TargetFrameworkVersion>
    </PropertyGroup>
    <Target Name="Test">
        <Message Importance="high" Text="[MSBuild]::IsTargetFrameworkCompatible('netcoreapp3.1', '$(TargetFramework)') = $([MSBuild]::IsTargetFrameworkCompatible('netcoreapp3.1', '$(TargetFramework)'))" />
        <Message Importance="high" Text="[MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp3.1') = $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'netcoreapp3.1'))" />
        <Message Importance="high" Text="[MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '3.1') = $([MSBuild]::VersionGreaterThanOrEquals('$(TargetFrameworkVersion)', '3.1'))" />
    </Target>
</Project>
```

```bash
$ dotnet build -f net461
[MSBuild]::IsTargetFrameworkCompatible('net461', 'netcoreapp3.1') = False

$ dotnet build -f netstandard2.0 # though, not possible for tests anyway
[MSBuild]::IsTargetFrameworkCompatible('netstandard2.0', 'netcoreapp3.1') = False

$ dotnet build -f netcoreapp3.1
[MSBuild]::IsTargetFrameworkCompatible('netcoreapp3.1', 'netcoreapp3.1') = True

$ dotnet build -f net5.0
[MSBuild]::IsTargetFrameworkCompatible('net5.0', 'netcoreapp3.1') = True

$ dotnet build -f net6.0
[MSBuild]::IsTargetFrameworkCompatible('net6.0', 'netcoreapp3.1') = True

$ dotnet build -f net7.0
[MSBuild]::IsTargetFrameworkCompatible('net7.0', 'netcoreapp3.1') = True
```

Reply:
I had no idea there was an `IsTargetFrameworkCompatible` - I'm going to file that away for the future.